honeybadgermpc/reed_solomon.py

```diff
@@ -313,37 +311,35 @@ def _robust_update(self):
                 break
 
             num_agreement = len(self._available_points) - len(errors)
-            if num_agreement >= self._min_points_required():
-                # Errors detected considered to be confirmed errors
-                self._confirmed_errors |= set(errors)
-                self._available_points -= set(errors)
-                error_indices = []
-                for e in errors:
-                    error_idx = self._z.index(e)
-                    error_indices.append(error_idx)
-                    del self._z[error_idx]
-                self._num_decoded += 1
-                self._available_data = self._available_data[1:]
-                self._partial_result.append(decoded)
-
-                for error_idx in error_indices:
-                    for i in range(len(self._available_data)):
-                        del self._available_data[i][error_idx]
-            else:
+            if num_agreement < self._min_points_required():
                 break
 
+            self._num_decoded += 1
+            self._available_data = self._available_data[1:]
+            self._partial_result.append(decoded)
+
+            # Errors detected considered to be confirmed errors
+            self._confirmed_errors |= set(errors)
+            self._available_points -= set(errors)
+
+            for e in errors:
+                error_idx = self._z.index(e)
+
+                del self._z[error_idx]
```

Comment:
NYC: Is this a list, how does delete from a list work? Does it create a new list? That's not good, can we not replace it with `None` and reuse the same list?

Reply:
[good explanation for dicts, i believe it's the same for lists](https://stackoverflow.com/a/36716057)
Essentially, this contains the overhead of removing the symbol from namespaces, but doesn't actually deallocate memory (unless this was the final remaining reference to the object, in which case I think it will get garbage collected). 
Setting this to None would likely have the same affect, and be faster.

Reply:
[turns out it's the fastest way to remove an element from a dirt](https://stackoverflow.com/questions/11277432/how-to-remove-a-key-from-a-python-dictionary)